### PR TITLE
[components/concepts] Add page for process instance modification

### LINKED ISSUE #1122: I can read about the process instance modification as a concept
## Description

With Zeebe `8.1`, we added a new feature to modify a process instance (https://github.com/camunda/zeebe/issues/9570). This is a complex feature and it is important that users understand the concept.

We should document the concept behind the feature, similar to other concepts like [Process instance creation](https://docs.camunda.io/docs/components/concepts/process-instance-creation/).

Related docs from Camunda Platform 7: https://docs.camunda.org/manual/7.17/user-guide/process-engine/process-instance-modification/

### PR DESCRIPTION
**Pull request by saig0**
## What is the purpose of the change

Explain the concept of process instance modification.

The PR adds the documentation for the `next` and the `8.1` version. 

closes #1122 

## Are there related marketing activities

The process instance modification was a new feature of the Camunda Platform version `8.1`.

## When should this change go live?

As soon as possible. The feature is available since Camunda Platform version `8.1`. 

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/process-instance-modification.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+The process instance completed the first task `A` and waits on task `B`. Task `C` is connected to task `A` by a
+non-interrupting message catch event. An external system should publish the message, but it is not available.
+
+To correct the state of the process instance, we modify it and activate task `C`. As a result, task `C` is active, and a
+job worker can pick it up.
+
+The process instance activates the element in the same way as the regular flow; for example, if the incoming sequence
+flow of the element would be taken. The activation of the element can include the following steps:
+
```
Comment (christinaausley) [inline docs/components/concepts/process-instance-modification.md:61]: Are the tabs here steps to complete in order? If so, I would instead list this in numerical order:
1. Apply the input variable mappings.
2. Create the event subscriptions; for example, of boundary events.
3. Apply additional logic depending on the element; for example, create a job for a service task.

**Review by christinaausley — APPROVED** [review]
One non-blocking comment and a few adjustments in my latest commit 👍  Let me know if you have any questions.

File: docs/components/concepts/process-instance-modification.md (hunk 1 above)
Comment (saig0) [reply docs/components/concepts/process-instance-modification.md:61]: No. The order of the steps is not important for the docs. :+1:

**Comment by akeller** [comment]
@remcowesterhoud are you able to review?

**Comment by remcowesterhoud** [comment]
Yes I will. I was a bit swarmed but I will try to review this tomorrow!

File: docs/components/concepts/process-instance-modification.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+and rejects the command.
+
+:::note
+The process instance can't terminate the last active element instance of a child process instance. As a result, the
+process instance doesn't apply the termination instruction and rejects the command.
+
+Instead, we can terminate the call activity that created the child process instance.
+:::
+
+## Execute the modification instructions
```
Comment (remcowesterhoud) [inline docs/components/concepts/process-instance-modification.md:175]: ❓ Do we think this is interesting knowledge for users? I feel like the scenarios have been explained above. The order in which we execute these instructions might be irrelevant.

File: docs/components/concepts/process-instance-modification.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+  - A start event of a process or a subprocess
+  - A boundary event
+  - An event that belongs to an event-based gateway
+  - An element inside a multi-instance subprocess
+  - A sequence flow
+- If the activating element is [a nested element](#activate-a-nested-element) and the subprocess has more than one
+  active instance.
+- If the modification terminates all active instances of a child process instance.
+
+## Use at your own risk
```
Comment (remcowesterhoud) [inline docs/components/concepts/process-instance-modification.md:208]: 👍 I like this section!

**Review by remcowesterhoud — APPROVED** [review]
Very well written @saig0 👍 

I have one question but nothing significant.

File: docs/components/concepts/process-instance-modification.md (hunk 2 above)
Comment (saig0) [reply docs/components/concepts/process-instance-modification.md:175]: I think that the section is important. It describes two parts:
- the transactional behavior
- the order of the instructions

Both parts are more advanced topics. They are maybe not interesting for users in the beginning. But for advanced users (e.g. users that are aware of the C7 behavior) or as a reference for us when we need to explain the behavior.

For example, to answer the question "Why I don't see a token if I cancel a subprocess and add a new token inside the subprocess?".

@remcowesterhoud does this make sense to you? Should we improve the section?

**Pull request merged**